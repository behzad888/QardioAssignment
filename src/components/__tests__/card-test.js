//@flow
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from '../card/card';

describe('while card test running in a browser environment', () => {
  let container, wrapper;

  beforeEach(() => {
    jest.resetModules();

    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    ReactDOM.render(null, container);
    document.body.removeChild(container);
    container = null;
  });

  it('should render correctly', () => {
    wrapper = renderer
      .create(
        <div>
        <Card> This is a Card View </Card>
        <Card className="card-test"> This is a Card View </Card>
        <Card className="card-test">
          <div>
            This is a Cord View with Card body
            </div>
        </Card>
        <Card className="card-test" header={<span>Header</span>}>
          <div>
            This is a Cord View with Card body
            </div>
        </Card>
        </div>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});