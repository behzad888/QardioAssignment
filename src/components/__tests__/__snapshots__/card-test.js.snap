// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while card test running in a browser environment should render correctly 1`] = `
<div>
  <div
    className="card "
  >
    <div
      className="card-body"
    >
       This is a Card View 
    </div>
  </div>
  <div
    className="card card-test"
  >
    <div
      className="card-body"
    >
       This is a Card View 
    </div>
  </div>
  <div
    className="card card-test"
  >
    <div
      className="card-body"
    >
      <div>
        This is a Cord View with Card body
      </div>
    </div>
  </div>
  <div
    className="card card-test"
  >
    <div
      className="card-header"
    >
      <span>
        Header
      </span>
    </div>
    <div
      className="card-body"
    >
      <div>
        This is a Cord View with Card body
      </div>
    </div>
  </div>
</div>
`;
